platform/core/src/utils/loadAndCacheDerivedDisplaySets.js

```diff
@@ -108,6 +116,7 @@ async function loadAndCacheDerivedDisplaySets(referencedDisplaySet, studies, log
 
     try {
       await recentDisplaySet.load(referencedDisplaySet, studies);
+      refreshViewports();
```

Comment:
So the thing is... 

The core package should contain business logic and we should strive as much as possible to align with the current architecture and avoid as much as possible to reference specific libraries here. 

Cornerstone code should live in cornerstone extension, and if this is specific to segmentation, we can trigger an event after loading and caching the derived display set, and then cs extension could refresh by subscribing to this event.

Reply:
I see your point. We actually have already a `extensiondicomsegmentationsegloaded` event everytime a seg is loaded. Thie event is used in the segmentation menu and a `refreshSegmentations` slot method is called. 

The problem is that when you just open the study, the segmentation menu is closed and therefore refreshSegmentations method is never called.

Reply:
at the moment there are some issues realted to the `refreshViewports()` (essentially `cornerstone.updateImage`). 

For segmentation
1) `getSourceDisplaySet` is called in many points, and `getSourceDisplaySet` calls for `setActiveLabelmap` (unless there is the specific flag `activateLabelMap` set to false). `setActiveLabelmap` actually also calls refreshViewports.
2) `getSourceDisplaySet` is used in the core. In `setViewportData` in ViewerMain.js and `mapDispatchToProps` in ConnectedStudyBrowser. And finally we have also the `load` in loadAndCacheDerivedDisplaySets (which needs refresh and triggered this discussion).
3) In extension, in the segmentaionPanel both  refreshViewports() and setActiveLabelmap are called in several palces.

In general cornerstone extension specific code is used both in the core and extensions, i.e.:
![image](https://user-images.githubusercontent.com/7985338/118497132-1477ca80-b725-11eb-901c-340f2cca0065.png)